src/solver/conv_asm_1x1u.cpp

```diff
@@ -254,6 +256,51 @@ bool PerformanceConfigConvAsm1x1U::IsValidValue() const
         && IsTwoPower<1,8>(waves_k_in_group); // clang-format on
 }
 
+bool PerformanceConfigConvAsm1x1U::IsValidValueDynamic() const
```

Comment:
This function is a partial duplicate of `IsValidValue()`.

💡 Create private IsValidValueImpl(..., bool allow_negative). IsValidValue calls it with false, IsPartiallyValidValue calls it with true.

Reply:
Instead of a bool to allow negative I'm going to pass in an integer value which is the length of the sequence predicted so far and since the model always predicts the parameters in a certain order I can be sure of what parameters have been set already so I can test the next parameter for validity. For default non-dynamic functionality I will always pass in 8 (total number of tuning parameters), as it assumes they have all been set.

Reply:
Good! Resolved.